library/core/src/sync/atomic.rs

```diff
@@ -2988,7 +3002,14 @@ atomic_int_ptr_sized! {
 }
 
 #[inline]
-#[cfg(target_has_atomic = "8")]
+#[cfg(any(
+    target_has_atomic = "8",
+    target_has_atomic = "16",
+    target_has_atomic = "32",
+    target_has_atomic = "64",
+    target_has_atomic = "128",
+    target_has_atomic = "ptr"
+))]
```

Comment:
I think that here https://github.com/rust-lang/rust/blob/893938f64f7a9ff531b3968c9a9e79e0524606bd/compiler/rustc_session/src/config.rs#L942 you could add before the loop:

```rust
if atomic_cas {
    ret.insert((sym::target_has_atomic, None));
}
```

And then I think that would make `#[cfg(target_has_atomic))` work? And the same could be done for _load_store.

Reply:
Done in a slightly different way (including `target_has_atomic_load_store` and doing that in the loop. PTAL